processing/src/main/java/io/druid/query/groupby/epinephelinae/BufferHashGrouper.java

```diff
@@ -58,10 +61,12 @@ public BufferHashGrouper(
       final AggregatorFactory[] aggregatorFactories,
       final int bufferGrouperMaxSize,
       final float maxLoadFactor,
-      final int initialBuckets
+      final int initialBuckets,
+      final boolean useDefaultSorting
```

Comment:
This parameter is confusingly named since it raises the question: which non-default sorting is used if `useDefaultSorting` is false?

A similar variable seems to be called `sortHasNonGroupingFields` in LimitedBufferHashGrouper. Would it make sense to use the same one here? It would at least be more consistent.

Reply:
I decided to keep it the same name but added some comments on its meaning, it's there t handle the case where some machines don't have enough buffer capacity to apply limit push down but other machines do.

It's not quite the same as sortHasNonGroupingFields, useDefaultSorting would be true when:
- Not using BufferHashGrouper for limit push down
- Using BufferHashGrouper because we don't have buffer capacity, but the sorting order has fields that are not in the grouping key, so we use the default ordering instead

False when
- Using BufferHashGrouper because we don't have buffer capacity for limit push down, but the OrderBy's ordering fields are all contained in the grouping key

Reply:
OK, thanks for adding the comment.